javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -104,9 +132,11 @@ private static int treeSize(int branchCount, int depthShift) {
         } else if (n >= length()) {
             return empty();
         } else {
-            final NodeModifier drop = Arrays::copyDrop;
-            final Object[] root = modifyLeaf(array, depthShift, offset + n, drop, drop);
-            return new BitMappedTrie<>(root, offset + n, length() - n, depthShift);
+            final int index = offset + n;
+            final Object[] root = arePointingToSameLeaf(0, n)
+                                  ? array
+                                  : modifyLeaf(array, depthShift, index, Arrays::copyDrop, Identity);
+            return new BitMappedTrie<T>(root, index, length() - n, depthShift).collapseHeight();
```

Comment:
Lorinc, could you please explain to me the internal meaning of `compactHeight()`. I understand that unnecessary parts of the BMT are released. But which modification of the BMT made it necessary? Does it make sense for the initial/minimal BMT also or do we need it because we introduced some sort of change to the BMT?

Thanks for 'sharpening me up' (in german: 'Aufschärfen') ;-)